src/api.ts

```
@@ -20,9 +20,31 @@ function get<T>(url: string): Promise<T> {
   // we add some delay to see how the loader works
   return wait(300)
     .then(() => fetch(fullURL))
-    .then(res => res.json());
+    .then(res => res.json())
+    .catch(error => {
+      window.alert(error);
+
+      return [];
+    });
```

It's generally not a good practice to return an empty array in the `catch` block of a Promise. This can mask errors and make debugging more difficult. Instead, you should re-throw the error or return a rejected Promise.

```typescript
.catch(error => {
  window.alert(error);
  throw error; // or return Promise.reject(error);
});
```

---

I don't get the data and don't render it accordingly, but I don't break the whole application by returning an undefined or just leaving it without a return. I display the error separately.